oak-benchmarks-lucene/src/main/java/org/apache/jackrabbit/oak/benchmark/util/OakLuceneIndexUtils.java

```diff
@@ -0,0 +1,106 @@
+     * @param type the types of the properties in order of the properties
+     * @param orderedPropsMap the ordered props and its properties
+     * @param persistencePath the path if the persistence=file (default is repository)
+     * @return the node just created
+     * @throws RepositoryException the repository exception
+     */
+    public static Node luceneIndexDefinition(Session session, String indexDefinitionName,
+                                             String async, String[] propertyNames, String[] type,
+                                             Map<String, Map<String, String>> orderedPropsMap, String persistencePath)
+            throws RepositoryException {
+
+        Node root = session.getRootNode();
+        Node indexDefRoot = JcrUtils.getOrAddNode(root, IndexConstants.INDEX_DEFINITIONS_NAME,
+                NodeTypeConstants.NT_UNSTRUCTURED);
+
+        Node indexDef = JcrUtils.getOrAddNode(indexDefRoot, indexDefinitionName,
+                IndexConstants.INDEX_DEFINITIONS_NODE_TYPE);
+
+        indexDef.setProperty(IndexConstants.TYPE_PROPERTY_NAME, LuceneIndexConstants.TYPE_LUCENE);
+        indexDef.setProperty(FulltextIndexConstants.FULL_TEXT_ENABLED, false);
+        if (async != null) {
+            indexDef.setProperty(IndexConstants.ASYNC_PROPERTY_NAME, async);
+        }
+        // Set indexed property names
+        indexDef.setProperty(FulltextIndexConstants.INCLUDE_PROPERTY_NAMES, propertyNames,
+                PropertyType.NAME);
+
+        Node propsNode = JcrUtils.getOrAddNode(indexDef, FulltextIndexConstants.PROP_NODE);
+        for (int i = 0; i < propertyNames.length; i++) {
+            Node propNode =
+                    JcrUtils.getOrAddNode(propsNode, propertyNames[i], NodeTypeConstants.NT_OAK_UNSTRUCTURED);
+            propNode.setProperty(FulltextIndexConstants.PROP_TYPE, type[i]);
+        }
+
+        // Set ordered property names
+        if ((orderedPropsMap != null) && !orderedPropsMap.isEmpty()) {
+            List<String> orderedProps = Lists.newArrayList();
+            for (Map.Entry<String, Map<String, String>> orderedPropEntry : orderedPropsMap
+                    .entrySet()) {
+                Node propNode = JcrUtils.getOrAddNode(propsNode, orderedPropEntry.getKey(),
+                        NodeTypeConstants.NT_OAK_UNSTRUCTURED);
+                propNode.setProperty(FulltextIndexConstants.PROP_TYPE,
+                        orderedPropEntry.getValue().get(FulltextIndexConstants.PROP_TYPE));
+                orderedProps.add(orderedPropEntry.getKey());
+            }
+            if (!orderedProps.isEmpty()) {
+                indexDef.setProperty(FulltextIndexConstants.ORDERED_PROP_NAMES,
+                        orderedProps.toArray(new String[orderedProps.size()]),
+                        PropertyType.NAME);
+            }
+        }
+
+        // Set file persistence if specified
+        if (!Strings.isNullOrEmpty(persistencePath)) {
```

Comment:
I don't think "persistence" is needed any longer. I would remove all three (PERSISTENCE_PATH, PERSISTENCE_NAME, PERSISTENCE_FILE) and the respective parameter

Reply:
@thomasmueller  - 
In the scalability tests , I see there are specific cases where this is being used - 
```
case LUCENE_FILE:
                persistencePath =
                    "target" + StandardSystemProperty.FILE_SEPARATOR.value() + "lucene" + String
                        .valueOf(System.currentTimeMillis());
                OakLuceneIndexUtils.luceneIndexDefinition(session, "customIndex", ASYNC_INDEX,
                        new String[]{FILTER_PROP, DATE_PROP},
                        new String[]{PropertyType.TYPENAME_STRING, PropertyType.TYPENAME_DATE},
                        null, persistencePath);
                break;
            case LUCENE_FILE_DOC:
                persistencePath =
                    "target" + StandardSystemProperty.FILE_SEPARATOR.value() + "lucene" + String
                        .valueOf(System.currentTimeMillis());
```

I think let's keep it as it is for now ...when I would be setting up all these tests in jenkins, then I can revisit and see if we can safely remove it. Wdyt ?